lib/samson/hooks.rb

```diff
@@ -80,6 +124,16 @@ def hooks(name)
 
 Dir["plugins/*/lib"].each { |f| $LOAD_PATH << f } # treat included plugins like gems
 
-Samson::Hooks.plugins.
-  each(&:require).
-  each(&:add_migrations)
+module LoadDecorators
+  def inherited(subclass)
+    Samson::Hooks.fire(:model_defined, subclass)
+    super
+  end
+end
+
+Samson::Hooks.callback :model_defined do |model_class|
```

Comment:
Preloading a decorators index (indexed by classname) when we run the plugin setup. Decorators will still be loaded on the ```model_defined``` callback.